src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs

```diff
@@ -339,15 +339,16 @@ private void OnRunCompleted(object sender, EventArgs e)
             _runComplete.Set();
         }
 
-        private int CountTestCases(ITest test, ITestFilter filter)
+        private static int CountTestCases(ITest test, ITestFilter filter)
         {
             if (!test.IsSuite)
                 return filter.Pass(test) ? 1: 0;
 
             int count = 0;
-            foreach (ITest child in test.Tests)
+            var tests = test.Tests;
```

Comment:
Tests is interface distpatch to get a collection which shouldn't change, calling only once

Reply:
This code is operating only on the test (ITest ).  Shouldn't it then be moved to the class implementing that?

Reply:
That would require adding a new member to `ITest` which would also take filter to allow counting

Reply:
Yes, and ITestFilter is also a public interface, so that would be fine I think.